### PR TITLE
Switch sdk_xcode_harness to use flutter build aot

### PR DESCRIPTION
**Pull request by abarth**
This patch moves a bunch of complexity out of FlutterApplication.xcodeproj and
into `flutter build aot`, which will make it easier for developers to create
their own FlutterApplication.xcodeproj.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde @tvolkert

File: sky/build/sdk_xcode_harness/Tools/common/BuildFlutterApp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  fi
+
+  AssertExists $1
+  local project_path=$1
+
+  local derived_dir=${SOURCE_ROOT}/FlutterApplication/Generated
+  RunCommand mkdir -p $derived_dir
+  AssertExists $derived_dir
+
+  rm -f ${derived_dir}/app.a
```
Comment (chinmaygarde) [inline sky/build/sdk_xcode_harness/Tools/common/BuildFlutterApp]: Decorate this with the RunCommand maybe.

**Comment by chinmaygarde** [comment]
lgtm

**Comment by abarth** [comment]
@chinmaygarde This seemed to work in my local testing.  Do you want me to land this or do you want to try it out a bit in person tomorrow first?

**Comment by chinmaygarde** [comment]
You can land it. Will test it again tomorrow.

**Pull request merged**